apps/service_gather/lib/gather/extraction/source_stream.ex

```diff
@@ -8,41 +8,104 @@ defmodule Gather.Extraction.SourceStream do
     """
     use Source.Handler
     use Properties, otp_app: :service_gather
+    use Annotated.Retry
+    require Logger
+
+    @max_tries get_config_value(:max_tries, default: 10)
+    @initial_delay get_config_value(:initial_delay, default: 500)
 
     getter(:dlq, default: Dlq)
+    getter(:app_name, required: true)
 
-    def handle_batch(batch, context) do
-      send(context.assigns.pid, {:source_batch, batch})
-      :ok
+    @impl Source.Handler
+    def handle_batch(batch, %{assigns: %{extract: extract}} = context) do
+      Decoder.decode(extract.decoder, [batch])
```

Comment:
Decoder was written to handle a stream of chunks, doesn't seem like this is true any more, should we refactor decoder to just handle a batch as well?

Reply:
yes? do we see a situation in which we would want decoder to be handling a stream at all given the problem that brought us here?
also, who is _we_ in this scenario refactoring Decoder?

Reply:
I don't think so, I think with the new source protocol, it is unlikely we would go back to a stream implementation

Reply:
Implementing in a different card.